### PR TITLE
feat(package): Support package version from setup.cfg (python).

### LINKED ISSUE #2886: Show package version for python packages with a `setup.cfg`
#### Is your feature request related to a problem? Please describe.

Not really a *problem* just a pure enhancement that would make starship **even** better for me!

I am primarily a python developer. I have projects that use poetry for their build tool and ones that use the standard library setup whereby a `setup.cfg` file is involved.

When developing on a poetry managed package, the version of the package is specified in a `pyproject.toml` file in the package root directory like so:

``` toml
[tool.poetry]
name = "my_cool_project"
version = "0.7.1"
```

Starship handles this nicely as shown below:

image

When developing against a setuptools managed packaged (the defacto standard for python), the version the package is specified in a `setup.cfg` file in the package root directory like so:

``` cfg
[metadata]
name = my_cool_project
version = 0.2.7
```

However starship currently doesn't handle this for me and the nice little package icon doesn't show up:

image

#### Describe the solution you'd like

I would love it if starship handled both of these cases identically, showing the package icon with the version in both cases.

#### Describe alternatives you've considered

I've looked at the Package Version configuration options and can't see a `setup.cfg` implementation there. I'm not sure if there are any alternative ways of doing this short of implementing this support.

#### Note on python packaging

I'm aware python has a much more complex packaging story than for example rust, with a variety of different files involved in the past. However, the community (at least using the python provided build tools) seems to now be stabilising on the presence of a `setup.cfg` which details the package metadata, and a `pyproject.toml` which specifies the tooling used to build the package for distribution.

There is a provisionally accepted PEP (Python Enhancement Proposal): [PEP621](https://www.python.org/dev/peps/pep-0621/) which will move to storing all project metadata inside `pyproject.toml` but things tend to move quite slowly in the python space (for good reason obviously) so I doubt this will be the officially accepted standard for some time.

This is all to say that I think if starship could support also using `setup.cfg` for the package version icon, I suspect this feature would remain valid for many years to come.

I am learning Rust at the moment but I've only just started and have absolutely no idea how to go about implementing this so I thought I would raise it here in the hopes you guys might take it on!

Edit: I've actually just looked into this a bit scanning through the codebase and it doesn't look like it would be *that* complicated to implement so I might take a crack at it after all! 🤓 

Thanks for the project, I absolutely love starship! Please let me know if I can be of any additional help for this issue!

### PR DESCRIPTION
**Pull request by FollowTheProcess**
#### Description

Add an additional package version extraction function to parse
the 'version' attribute under the 'metadata' section in a python
package 'setup.cfg' file.

The implementation was largely based on the very similar
`extract_poetry_version` function.

Also add similar tests from the poetry extraction function to test
the desired behaviour.

#### Motivation and Context

This feature would enable starship to display the package version from a python package
`setup.cfg` file (the defacto standard).

Closes #2886 

#### How Has This Been Tested?

I followed the contributing guidelines and ran

`cargo clippy --all-targets --all-features`

and ran `cargo test`

I added two additional tests to cover the new function I added which are based heavily on the tests for
the poetry equivalent function, which are both passing locally.

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly. 
- [x] I have updated the tests accordingly.

I think the documentation will need updating to include a mention to `setup.cfg` in the packages module which I am more than happy to do, just want to submit this and get someone who knows way more about rust than I do to check it over (I am very new to rust).

Thanks!

### CODE REVIEW

**Comment by FollowTheProcess** [comment]
Sorry I seem to have failed the semantic PR workflow, but I included `feat` in the title? Is this because my commit message did not have `feat` in it?

**Comment by davidkna** [comment]
Can you use `rust-ini` instead of `configparser`? Starship alraedy depends on a parser for ini files.

**Comment by davidkna** [comment]
> Sorry I seem to have failed the semantic PR workflow, but I included `feat` in the title? Is this because my commit message did not have `feat` in it?

If you only have a single commit its message needs to follow the semantic guidelines and include `feat:`. For multiple commits changing the PR title is enough.

**Comment by FollowTheProcess** [comment]
> Can you use `rust-ini` instead of `configparser`? Starship already depends on a parser for ini files.

Yeah sure I can do that, I also just noticed I left an unnecessary comment in which I've just removed in an additional commit

**Review by cymruu — APPROVED** [review]
LGTM 👌

**Comment by FollowTheProcess** [comment]
@cymruu Do you want me to put a mention to this in the docs?

**Comment by cymruu** [comment]
@FollowTheProcess because other version parsers are described in `docs/config/README.md` I think that this one should be described too. But please wait for maintainers reviews. 
https://github.com/starship/starship/blob/master/docs/config/README.md#package-version

**Review by matchai — COMMENTED** [review]
Looks good to me 👍 
We should be ready to merge it as soon as the docs are updated to reflect the change 😊

**Comment by FollowTheProcess** [comment]
Cool, I dropped a mention in the docs, let me know if you need anything else! 🙂

**Review by andytom — APPROVED** [review]
LGTM

**Comment by andytom** [comment]
Thank you for your contribution @FollowTheProcess.

**Pull request merged**
